### PR TITLE
Log files:scan exception, add InterruptedException

### PR DESCRIPTION
**Pull request by MorrisJobke**
* downstream of https://github.com/owncloud/core/pull/27209
* tested and works 👍

### CODE REVIEW

File: apps/files/lib/Command/Scan.php [hunk 1]
```diff
@@ -161,11 +162,12 @@ protected function scanFiles($user, $path, $verbose, OutputInterface $output, $b
...
 			$output->writeln("<error>Home storage for user $user not writable</error>");
 			$output->writeln("Make sure you're running the scan command only as the user the web server runs as");
-		} catch (\Exception $e) {
-			if ($e->getMessage() !== 'ctrl-c') {
-				$output->writeln('<error>Exception while scanning: ' . $e->getMessage() . "\n" . $e->getTraceAsString() . '</error>');
-			}
+		} catch (InterruptedException $e) {
+			# exit the function if ctrl-c has been pressed
+			$output->writeln('Interrupted by user');
 			return;
```
Comment (ChristophWurst) [inline apps/files/lib/Command/Scan.php]: useless return, the functions ends anyway

File: apps/files/lib/Command/Scan.php [hunk 2]
```diff
@@ -161,11 +162,12 @@ protected function scanFiles($user, $path, $verbose, OutputInterface $output, $b
...
+		} catch (\Exception $e) {
+			$output->writeln('<error>Exception during scan: ' . $e->getMessage() . "\n" . $e->getTraceAsString() . '</error>');
```
Comment (ChristophWurst) [inline apps/files/lib/Command/Scan.php]: could use PHP_EOL instead of \n, but since we only support Linux this is fine I assume.

**Review by ChristophWurst — APPROVED** [review]
Some minor nitpicks, but the rest looks good

**Review by rullzer — APPROVED** [review]

**Comment by MorrisJobke** [comment]
@ChristophWurst I addressed your comments.

**Review by LukasReschke — CHANGES_REQUESTED** [review]
Needs autoloader update. Will do.

**Review by LukasReschke — APPROVED** [review]

**Comment by MorrisJobke** [comment]
:+1:

**Comment by MorrisJobke** [comment]
Autoloader test runs fine locally -> merge

**Pull request merged**